lib/ocpp/v201/smart_charging.cpp

```diff
@@ -621,4 +649,61 @@ SmartChargingHandler::verify_no_conflicting_external_constraints_id(const Chargi
     return result;
 }
 
+CompositeSchedule SmartChargingHandler::calculate_composite_schedule(
+    std::vector<ChargingProfile>& valid_profiles, const ocpp::DateTime& start_time, const ocpp::DateTime& end_time,
+    const int32_t evse_id, std::optional<ChargingRateUnitEnum> charging_rate_unit) {
+
+    std::optional<ocpp::DateTime> session_start{};
+
+    if (this->evse_manager.does_evse_exist(evse_id) and
+        this->evse_manager.get_evse(evse_id).get_transaction() != nullptr) {
```

Comment:
This can lead to an EvseOutOfRange for evse_id == 0, so I suggest: 
```suggestion
    if (this->evse_manager.does_evse_exist(evse_id) and evse_id != 0 and
        this->evse_manager.get_evse(evse_id).get_transaction() != nullptr) {
```